planex/cmd/makesrpm.py

```diff
@@ -136,14 +129,20 @@ def populate_working_directory(tmpdir, spec):
         print("The following archives have been ignored: {}".format(skipped))
 
     newspec = os.path.join(tmpdir, os.path.basename(spec.specpath()))
-    manifests = {}
+    manifests = {
+        name: sha
+        for name, sha in [
+            extract_commit(source[0]) for source in spec.sources()
+        ]
+        if name is not None
+    }
+    srpm_sources = sources if metadata else None
 
     with open(newspec, "w") as out:
-        out.writelines(spec.rewrite_spec())
+        out.writelines(
+            spec.rewrite_spec(srpm_sources=srpm_sources, manifests=manifests))
 
-    if manifests:
-        add_manifest_entry(manifests, newspec)
-    else:
+    if not manifests:
```

Comment:
We probably don't need this anymore if we're not strictly limited to reading the SHA from the .gitarchive-info.